clang-tools-extra/clang-tidy/modernize/UseDigitSeparatorCheck.cpp

```diff
@@ -0,0 +1,235 @@
+                      SplittedIntegerLiteral.end(), std::string(""),
+                      [](std::basic_string<char> S1,
+                         std::basic_string<char> S2) { return S1 + "\'" + S2; })
+          .erase(0, 1) +
+      Postfix;
+
+  return FormatedLiteralString;
+}
+
+std::string getFormatedFloatString(const llvm::StringRef OriginalLiteralString,
+                                   const llvm::APFloat FloatValue) {
+  if (OriginalLiteralString.contains('E') ||
+      OriginalLiteralString.contains('e')) {
+    return getFormatedScientificFloatString(OriginalLiteralString);
+  }
+
+  // Configure formatting
+  std::string Postfix;
+  for (const char &Character : OriginalLiteralString) {
+    if (!std::isdigit(Character) && Character != '.' && Character != '\'') {
+      Postfix += Character;
+    }
+  }
+
+  // Get precision
+  const std::string::size_type OriginalDotPosition =
+      OriginalLiteralString.find('.');
+  const llvm::StringRef OriginalFractionalSubString =
+      OriginalLiteralString.substr(OriginalDotPosition + 1,
+                                   OriginalLiteralString.size());
+  int Precision = 0;
+  for (const char &Character : OriginalFractionalSubString) {
+    if (std::isdigit(Character)) {
+      Precision++;
+    }
+  }
+
+  // Get formatting literal text
+
+  // Get string representation of float value
+  std::ostringstream StringStream;
+  StringStream << std::fixed << std::setprecision(Precision)
+               << FloatValue.convertToDouble();
+  const std::string FloatString = StringStream.str();
+
+  // Get integer and fractional parts of float number
+  const std::string::size_type DotPosition = FloatString.find('.');
+  const std::string IntegerSubString = FloatString.substr(0, DotPosition);
+  std::string FractionalSubString =
+      FloatString.substr(DotPosition + 1, FloatString.size());
+  std::reverse(FractionalSubString.begin(), FractionalSubString.end());
+
+  // Get formatting literal text
+  const std::string FormatedIntegerSubString = getFormatedIntegerString(
+      IntegerSubString, llvm::APInt(128, std::stoll(IntegerSubString)));
+  std::string FormatedFractionalSubString = getFormatedIntegerString(
+      FractionalSubString, llvm::APInt(128, std::stoll(FractionalSubString)));
+  std::reverse(FormatedFractionalSubString.begin(),
+               FormatedFractionalSubString.end());
+
+  const std::string FormatedLiteralString =
+      FormatedIntegerSubString + '.' + FormatedFractionalSubString + Postfix;
+
+  return FormatedLiteralString;
+}
+
+std::string
+getFormatedScientificFloatString(const llvm::StringRef OriginalLiteralString) {
+  // Split string to mantissa and exponent
+  const char EChar =
+      OriginalLiteralString.str().find('E') != std::string::npos ? 'E' : 'e';
+  const std::string::size_type EPosition =
+      OriginalLiteralString.str().find(EChar);
+  const std::string SignSymbol =
+      OriginalLiteralString[EPosition + 1] == '-'
```

Comment:
what if OriginalLiteralString does not have 'e' character, or it's last and then here we could have buffer overflow